iexec-sms-library/src/main/java/com/iexec/sms/api/SmsClient.java

```diff
@@ -90,14 +94,29 @@ ApiResponseBody<TeeSessionGenerationResponse, TeeSessionGenerationError> generat
             WorkerpoolAuthorization workerpoolAuthorization
     );
 
-    @RequestLine("GET /tee/workflow/config")
-    TeeWorkflowConfiguration getTeeWorkflowConfiguration();
-
     @RequestLine("POST /untee/secrets")
     @Headers("Authorization: {authorization}")
     SmsSecretResponse getUnTeeSecrets(
             @Param("authorization") String authorization,
             WorkerpoolAuthorization workerpoolAuthorization
     );
 
+    @RequestLine("GET /tee/provider")
+    TeeEnclaveProvider getTeeEnclaveProvider();
+
+    @RequestLine("GET /tee/config/scone")
+    SconeServicesConfiguration getSconeServicesConfiguration();
+
+    @RequestLine("GET /tee/config/gramine")
+    GramineServicesConfiguration getGramineServicesConfiguration();
```

Comment:
How about:
```suggestion
    @RequestLine("GET /tee/config")
    SconeServicesConfiguration getSconeServicesConfiguration();

    @RequestLine("GET /tee/config")
    GramineServicesConfiguration getGramineServicesConfiguration();
```
? :)

Reply:
I remind we discussed that point before, agreeing `/tee/config` should be sufficient. However, to avoid issues, it came to myself that the SMS should check the client is asking for the right type of configuration. See below:
https://github.com/iExecBlockchainComputing/iexec-sms/blob/bb9c19276e2f66e69be873fa02c76c3ffefd0e29/src/main/java/com/iexec/sms/tee/TeeController.java#L95-L99*
To achieve that goal, it seems better to include `/{teeEnclaveProvider]` in the path. But I'm open for other proposals :)

Reply:
Yes, I've seen that, I also like the idea of verifying that the SMS is compatible, but I wonder where. Maybe the core should verify it (when asking SMS to generate enclave challenge)?

Reply:
FYI: here is a commit to introduce this check in the Core: https://github.com/iExecBlockchainComputing/iexec-core/pull/521/commits/488a03758e5951a5c0bdd8adfa57440f5dc1f865
We'll continue our discussions on whether we should keep this check at each call to `GET /tee/config`.